File: FoodOrganizer/app/src/main/java/es/fron99/foodorganize/Fragments/ActivityTotal/FragmentCalendarMenus.java
package es.fron99.foodorganize.Fragments.ActivityTotal;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.Calendar;

import devs.mulham.horizontalcalendar.HorizontalCalendar;
import devs.mulham.horizontalcalendar.utils.HorizontalCalendarListener;
import es.fron99.foodorganize.R;


public class FragmentCalendarMenus extends Fragment {

    public FragmentCalendarMenus() {}

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_calendar_menus, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        Calendar startDate = Calendar.getInstance();
        startDate.add(Calendar.DAY_OF_MONTH,0);
        Calendar endDate = Calendar.getInstance();
        endDate.add(Calendar.DAY_OF_MONTH, 6);

        HorizontalCalendar horizontalCalendar = new HorizontalCalendar.Builder(requireActivity(), R.id.calendarView)
                .range(startDate, endDate)
                .datesNumberOnScreen(5)
                .build();


        horizontalCalendar.centerCalendarToPosition(0);

        horizontalCalendar.setCalendarListener(new HorizontalCalendarListener() {
            @Override
            public void onDateSelected(Calendar date, int position) {
                if (date.get(Calendar.DAY_OF_MONTH) == Calendar.getInstance().get(Calendar.DAY_OF_MONTH)){
                }
            }
        });

        /*

        ArrayList<Menu> menusLunch = new ArrayList<>(Arrays.asList(new Menu(),new Menu(),new Menu(),new Menu()));
        ArrayList<Menu> menusDinner = new ArrayList<>(Arrays.asList(new Menu(),new Menu(),new Menu(),new Menu()));

        RecyclerView recycledLunch = view.findViewById(R.id.recycledLunch);
        RecyclerView recycledDinner = view.findViewById(R.id.recycledDinner);

        recycledDinner.setLayoutManager(new LinearLayoutManager(getContext(),LinearLayoutManager.HORIZONTAL, false));
        recycledDinner.setAdapter(new AdapterListMenus(menusDinner));

        recycledLunch.setLayoutManager(new LinearLayoutManager(getContext(),LinearLayoutManager.HORIZONTAL, false));
        recycledLunch.setAdapter(new AdapterListMenus(menusLunch));
*/

    }
}

File: FoodOrganizer/app/src/main/java/es/fron99/foodorganize/Fragments/ActivityTotal/FragmentCalendarMenus.kt
package es.fron99.foodorganize.Fragments.ActivityTotal

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import devs.mulham.horizontalcalendar.HorizontalCalendar
import devs.mulham.horizontalcalendar.HorizontalCalendarView
import devs.mulham.horizontalcalendar.utils.HorizontalCalendarListener
import es.fron99.foodorganize.Adapters.AdapterListCalendarMenus
import es.fron99.foodorganize.Models.Food
import es.fron99.foodorganize.Models.Menu
import es.fron99.foodorganize.Models.TimeMenu
import es.fron99.foodorganize.R
import es.fron99.foodorganize.ViewModels.ActivityTotalVM
import java.sql.Time
import java.util.*
import kotlin.collections.ArrayList

class FragmentCalendarMenus : Fragment() {

    private lateinit var activityTotalVM : ActivityTotalVM

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_calendar_menus, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        activityTotalVM = ViewModelProvider(requireActivity()).get(ActivityTotalVM::class.java)

        val startDate = activityTotalVM.daySelected
        startDate.add(Calendar.DAY_OF_MONTH, -7)
        val endDate = activityTotalVM.daySelected
        endDate.add(Calendar.DAY_OF_MONTH, 7)

        val horizontalCalendar = HorizontalCalendar.Builder(requireActivity(), R.id.calendarView)
                .range(startDate, endDate)
                .datesNumberOnScreen(5)
                .build()

        horizontalCalendar.centerCalendarToPosition(0)

        horizontalCalendar.calendarListener = object : HorizontalCalendarListener() {

            override fun onDateSelected(date: Calendar, position: Int) {
                activityTotalVM.daySelected = date
            }

            override fun onCalendarScroll(calendarView: HorizontalCalendarView?, dx: Int, dy: Int) {
                super.onCalendarScroll(calendarView, dx, dy)
            }

            override fun onDateLongClicked(date: Calendar?, position: Int): Boolean {
                activityTotalVM.daySelected = date
                return super.onDateLongClicked(date, position)
            }

        }


        val recycledLunch = view.findViewById<RecyclerView>(R.id.recycledTimeMenu)
        recycledLunch.layoutManager = LinearLayoutManager(context)
        recycledLunch.adapter = AdapterListCalendarMenus(context, activityTotalVM.timeMenus.value)

    }

}